### PR TITLE
Adding sonar support + infrastructure complete results

### PR DESCRIPTION
**Pull request by tiagobcx**

### CODE REVIEW

File: internal/commands/result.go [hunk 1]
```diff
@@ -322,6 +353,13 @@ func convertCxResultsToSarif(results *wrappers.ScanResultsCollection) *wrappers.
 	return sarif
 }
 
+func convertCxResultsToSonar(results *wrappers.ScanResultsCollection) *wrappers.ScanResultsSonar {
+	var sonar = new(wrappers.ScanResultsSonar)
+	sonar.Results = []wrappers.SonarIssues{}
```
Comment (diogopcx) [inline internal/commands/result.go:358]: is this assignment necessary, as the next line assigns to the same variable?

File: internal/commands/result.go [hunk 2]
```diff
@@ -322,6 +353,13 @@ func convertCxResultsToSarif(results *wrappers.ScanResultsCollection) *wrappers.
...
+	sonar.Results = createSonarRun(results)
```
Comment (diogopcx) [inline internal/commands/result.go:359]: createSonarRun returs parseResultsSonar
could we simplify this to sonar.Results = parseResultsSonar(results) and remove createSonarRun?

File: test/integration/result_test.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ func TestResultListJson(t *testing.T) {
 
 // assert all files were created
 func assertResultFilesCreated(t *testing.T) {
-	extensions := []string{util.FormatJSON, util.FormatSarif, util.FormatHTML}
+	extensions := []string{util.FormatJSON, util.FormatSarif, util.FormatHTML,util.FormatJSON}
```
Comment (diogopcx) [inline test/integration/result_test.go:47]: is this formatted?

File: internal/commands/result.go (hunk 1 above)
Comment (tiagobcx) [reply internal/commands/result.go:358]: It is not, will fix it

File: internal/commands/result.go (hunk 2 above)
Comment (tiagobcx) [reply internal/commands/result.go:359]: Yes it will be more simple

File: test/integration/result_test.go (hunk 3 above)
Comment (tiagobcx) [reply test/integration/result_test.go]: fixed

**Review by diogopcx — APPROVED** [review]

**Pull request merged**